### PR TITLE
ISPN-12138 be able to get with other key content types

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-12138

### CODE REVIEW

File: src/services/utils.ts [hunk 1]
```diff
@@ -3,18 +3,17 @@
...
 export enum KeyContentType {
-  OctetStream = 'application/octet-stream',
-  OctetStreamHex = 'application/octet-stream; encoding=hex',
   StringContentType = 'java.lang.String',
   DoubleContentType = 'java.lang.Double',
   IntegerContentType = 'java.lang.Integer',
   LongContentType = 'java.lang.Long',
   BooleanContentType = 'java.lang.Boolean',
-  BytesType = 'Bytes'
+  BytesType = 'Bytes',
```
Comment (gustavonalle) [inline src/services/utils.ts:14]: It'd be better to remove the 'Bytes' since it is covered by the other two below

File: src/services/utils.ts [hunk 2]
```diff
@@ -3,18 +3,17 @@
...
+  OctetStream = 'application/octet-stream',
```
Comment (gustavonalle) [inline src/services/utils.ts:15]: A less 'technical' way of present this is ```byte[] as Base64```

File: src/services/utils.ts [hunk 3]
```diff
@@ -3,18 +3,17 @@
...
+  OctetStreamHex = 'application/octet-stream; encoding=hex'
```
Comment (gustavonalle) [inline src/services/utils.ts:16]: Similar as above, ```byte[] as Hex```

File: src/services/utils.ts (hunk 1 above)
Comment (karesti) [reply src/services/utils.ts:14]: ok

File: src/services/utils.ts (hunk 3 above)
Comment (karesti) [reply src/services/utils.ts:16]: Which is the  list I should add ? from the console the octet-stream makes sense ?

Comment (karesti) [reply src/services/utils.ts:16]: and If I add a marshalled 'Person' protostream type, how should I deal it? is it possible ?

Comment (gustavonalle) [reply src/services/utils.ts:16]: It is not possible to add specific protobuf types to the key content type header. 

> Which is the list I should add 

The same list you have know, but removing ```Bytes``` and adding  ```byte[] as Base64``` and ```byte[] as Hex```

Comment (gustavonalle) [reply src/services/utils.ts:16]: > Which is the list I should add

You are sending the headers correctly, but the change I suggest is to display it differently (byte[] as Hex)  and (byte[] as Base64)

**Comment by karesti** [comment]
@gustavonalle I changed the display names. btw, if I want to add an Integer instead of a text in the value, how can I do it?

File: src/services/utils.ts (hunk 3 above)
Comment (karesti) [reply src/services/utils.ts]: great, done and pushed

File: src/services/utils.ts (hunk 2 above)
Comment (karesti) [reply src/services/utils.ts]: done

**Comment by gustavonalle** [comment]
I can't seen to make it work. 
Steps:
- Create a cache named 'test' with protobuf_dist template
- Add entry
- Key 1
- Value 1

 Error:

Invalid top level object! Found token: NUMBER

**Comment by gustavonalle** [comment]
There is an issue in the server that is causing the edit issues: https://issues.redhat.com/browse/ISPN-12315

**Comment by gustavonalle** [comment]
It needs ISPN-12315, I haven't got to it yet, plan to do it for 12.0.0.Final latest

**Comment by karesti** [comment]
at least the other parts can be integrated if it's working. WDYT ?

**Comment by gustavonalle** [comment]
> at least the other parts can be integrated if it's working. WDYT ?

yes

**Comment by karesti** [comment]
@gustavonalle please grab a look and tell me if it feels like mergeable now (even if some bugs on the server yet)

**Review by gustavonalle — APPROVED** [review]

**Comment by gustavonalle** [comment]
Merged, thanks @karesti !

**Pull request merged**